vimiv/imutils/metadata.py

```diff
@@ -0,0 +1,418 @@
+        """Fetch a single metadata key.
+
+        Args:
+            _base_key: metadata key to fetch.
+        """
+        self.raise_exception("Getting formatted keys")
+
+    def get_keys(self) -> Iterable[str]:
+        """Retrieve the name of all metadata keys available."""
+        self.raise_exception("Getting metadata keys")
+
+    @classmethod
+    def raise_exception(cls, operation: str) -> NoReturn:
+        """Raise an exception for a not implemented metadata operation."""
+        msg = f"{operation} is not supported{cls.MESSAGE_SUFFIX}"
+        _logger.warning(msg, once=True)
+        raise UnsupportedMetadataOperation(msg)
+
+
+class _ExternalKeyHandlerPiexif(_ExternalKeyHandlerBase):
+    """Implementation of ExternalKeyHandler based on piexif."""
+
+    MESSAGE_SUFFIX = " by piexif."
+
+    def __init__(self, filename=""):
+        super().__init__(filename)
+        try:
+            self._metadata = piexif.load(filename)
+        except FileNotFoundError:
+            _logger.debug("File %s not found", filename)
+            self._metadata = None
+
+    def fetch_key(self, base_key: str) -> Tuple[str, str, str]:
+        key = base_key.rpartition(".")[2]
+
+        try:
+            for ifd in self._metadata:
+                if ifd == "thumbnail":
+                    continue
+
+                for tag in self._metadata[ifd]:
+                    keyname = piexif.TAGS[ifd][tag]["name"]
+                    keytype = piexif.TAGS[ifd][tag]["type"]
+                    val = self._metadata[ifd][tag]
+                    _logger.debug(
+                        f"name: {keyname}\
+                        type: {keytype}\
+                        value: {val}\
+                        tag: {tag}"
+                    )
+                    if keyname != key:
+                        continue
+                    if keytype in (
+                        piexif.TYPES.Byte,
+                        piexif.TYPES.Short,
+                        piexif.TYPES.Long,
+                        piexif.TYPES.SByte,
+                        piexif.TYPES.SShort,
+                        piexif.TYPES.SLong,
+                        piexif.TYPES.Float,
+                        piexif.TYPES.DFloat,
+                    ):  # integer and float
+                        return (keyname, keyname, str(val))
+                    if keytype in (
+                        piexif.TYPES.Ascii,
+                        piexif.TYPES.Undefined,
+                    ):  # byte encoded
+                        return (keyname, keyname, val.decode())
+                    if keytype in (
+                        piexif.TYPES.Rational,
+                        piexif.TYPES.SRational,
+                    ):  # (int, int) <=> numerator, denominator
+                        return (keyname, keyname, f"{val[0]}/{val[1]}")
+
+        except (piexif.InvalidImageDataError, KeyError):
+            return ("", "", "")
+
+        return ("", "", "")
```

Comment:
The same thing happens here. I would rewrite this like this:
```python
with contextlib.suppress(piexif.InvalidImageDateError):
    # All the key handling here
    
raise KeyError(f"Key '{base_key}' not found")
```
so the metadata widget can ignore the key by catching the exception.

Reply:
Fixed in 16ec200